subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/catalog/VersionCatalogExtensionIntegrationTest.groovy

```diff
@@ -1847,4 +1847,46 @@ Second: 1.1"""
         outputContains "Found plugin: 'com.acme.greeter2:{require 1.0.0; prefer 1.1.0; reject 1.0.5}'."
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/17874")
+    def "support version catalogs in resolutionStrategy"() {
+        settingsFile << """
+            dependencyResolutionManagement {
+                versionCatalogs {
+                    libs {
+                        alias("myLib").to("org.gradle.test", "lib").version {
```

Comment:
I am just curios: any reason you decided to allow version catalog entries with prefer/strictly etc notations?

I will update this PR and forbid such entries since these versions already limit resolution

Reply:
Thanks for your review. I hesitated about this, and thought it's not harmful to support it, didn't realize that it violates the semantics/purpose of `force`.